tests/http/test_proxyagent.py

```diff
@@ -292,6 +294,134 @@ def test_https_request_via_proxy(self):
         body = self.successResultOf(treq.content(resp))
         self.assertEqual(body, b"result")
 
+    def test_http_request_via_proxy_with_blacklist(self):
+        # The blacklist includes the configured proxy IP.
+        agent = ProxyAgent(
+            BlacklistingReactorWrapper(
+                self.reactor, ip_whitelist=None, ip_blacklist=IPSet(["1.0.0.0/8"])
+            ),
+            self.reactor,
+            http_proxy=b"proxy.com:8888",
+        )
+
+        self.reactor.lookups["proxy.com"] = "1.2.3.5"
```

Comment:
Shouldn't this IP be in the blacklist range?

Reply:
It is, 1.2.3.5 is in 1.0.0.0/8

Reply:
:facepalm: I read the range as 10.*